08_project/kasunfdo/cmd/puppy-server/main_test.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	os.RemoveAll("./data")
```

Comment:
Looks like this is duplicated. You are performing the function call, then defering the function call which will call it again. Can you not just do this in defer?

Reply:
Thank you for pointing that out. 
I did it to remove data dir before test and after test. 
If I remove`defer os.RemoveAll("./data")`, it would keep test data. 
If I remove `os.RemoveAll("./data")`, test fails if program is already ran once.

Anyway, I had to use it for testing leveldb store and code coverage dropped due to error paths I couldn't test. So I removed leveldb implementation and these two lines are leftovers. :D

Reply:
@rokane, could you please approve?